builder/dockerfile/dispatchers.go

```diff
@@ -315,20 +315,18 @@ func workdir(req dispatchRequest) error {
 		return nil
 	}
 
-	cmd := req.runConfig.Cmd
-	comment := "WORKDIR " + req.runConfig.WorkingDir
-	// reset the command for cache detection
-	req.runConfig.Cmd = strslice.StrSlice(append(getShell(req.runConfig), "#(nop) "+comment))
-	defer func(cmd strslice.StrSlice) { req.runConfig.Cmd = cmd }(cmd)
+	// TODO: why is this done here. This seems to be done at random places all over
+	// the builder
+	req.runConfig.Image = req.builder.image
```

Comment:
These TODO comments about setting runConfig.Image from b.Image are addressed in a follow up PR #32749